### PR TITLE
Convert CountryDropdown to TypeScript

### PR DESCRIPTION
**Pull request by resynth1943**
Refactored; TypeScript; functional component.

Signed-off-by: Resynth <resynth1943@tutanota.com>

### CODE REVIEW

File: src/components/views/auth/CountryDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    /** The country code to use. */
+    value?: string;
+
+    /** Whether or not to append a numeral prefix to the value. */
+    isSmall?: boolean;
+
+    disabled?: true;
+}
+
+function countryMatchesSearchQuery(query, country) {
```
Comment (t3chguy) [inline src/components/views/auth/CountryDropdown.tsx:52]: missing param types

File: src/components/views/auth/CountryDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+CountryDropdown.propTypes = {
+    className: PropTypes.string,
+    isSmall: PropTypes.bool,
+    // if isSmall, show +44 in the selected value
+    showPrefix: PropTypes.bool,
+    onOptionChange: PropTypes.func.isRequired,
+    value: PropTypes.string,
+    disabled: PropTypes.bool,
+};
```
Comment (t3chguy) [inline src/components/views/auth/CountryDropdown.tsx:174]: we replace propTypes with Typescript interfaces

**Review by t3chguy — CHANGES_REQUESTED** [review]

File: src/components/views/auth/CountryDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (resynth1943) [inline src/components/views/auth/CountryDropdown.tsx:18]:
```suggestion
```

No `propTypes`.

File: src/components/views/auth/CountryDropdown.tsx (hunk 1 above)
Comment (resynth1943) [inline src/components/views/auth/CountryDropdown.tsx:52]:
```suggestion
function countryMatchesSearchQuery(query: string, country: Country) {
```

Add parameter types here.

File: src/components/views/auth/CountryDropdown.tsx (hunk 2 above)
Comment (resynth1943) [inline src/components/views/auth/CountryDropdown.tsx:174]:
```suggestion
```

Don't need `propTypes` here.

File: src/components/views/auth/CountryDropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    ));
+
+    return (
+        <Dropdown
+            id="mx_CountryDropdown"
+            className={props.className + " mx_CountryDropdown"}
+            onOptionChange={(iso2) => onOptionChange(COUNTRIES_BY_ISO2[iso2])}
+            onSearchChange={(search) => setSearchQuery(search)}
+            menuWidth={298}
+            getShortOption={iso2 => <ShortOption {...props} value={iso2} />}
```
Comment (resynth1943) [inline src/components/views/auth/CountryDropdown.tsx:155]:
```suggestion
            getShortOption={(iso2) => <ShortOption {...props} value={iso2} />}
```

For consistency.

**Review by resynth1943 — COMMENTED** [review]
Left some changes to apply (for myself).

File: src/components/views/auth/CountryDropdown.tsx (hunk 2 above)
Comment (resynth1943) [reply src/components/views/auth/CountryDropdown.tsx]: See 3d83a8b and d2a02e4.

**Comment by resynth1943** [comment]
Tests are passing. It's all yours :-)

File: src/components/views/auth/CountryDropdown.tsx (hunk 3 above)
Comment (t3chguy) [reply src/components/views/auth/CountryDropdown.tsx:18]: This isn't resolved.

File: src/components/views/auth/CountryDropdown.tsx (hunk 2 above)
Comment (t3chguy) [reply src/components/views/auth/CountryDropdown.tsx:174]: This isn't resolved.

Comment (resynth1943) [reply src/components/views/auth/CountryDropdown.tsx]: Sorry! See https://github.com/matrix-org/matrix-react-sdk/pull/5285/commits/bb99f8d18cd6849434538f8a36e74f463caa6fd7

**Comment by t3chguy** [comment]
You'll need to pull in latest develop to get the tests passing again

**Comment by jryans** [comment]
I believe the author is no longer engaged, so it makes sense to close this for now.

**Pull request closed without merging**